### PR TITLE
Add support for Covid Testing

### PR DESCRIPTION
**Pull request by BlairArchibald**
This allows Adults/Pensioners to get a test (around) a day after symptoms appear and use this to determine if they should remain in isolation/quarantine

### CODE REVIEW

File: src/main/java/uk/co/ramp/covid/simulation/covid/Covid.java [hunk 1]
```diff
@@ -206,9 +206,9 @@ public double getSymptomDelay() {
     public double getTransAdjustment() {
     	double transAdjustment = 0.0;
     	if(asymptomatic) transAdjustment = this.asymptomaticTransAdjustment;
-    	else if(phase1 && !this.getIsSymptomatic()) transAdjustment = this.asymptomaticTransAdjustment;
-    	else if(phase1 && this.getIsSymptomatic()) transAdjustment = this.symptomaticTransAdjustment;
-    	else if(phase2 && this.getIsSymptomatic()) transAdjustment = this.symptomaticTransAdjustment;
+    	else if(phase1 && !this.isSymptomatic()) transAdjustment = this.asymptomaticTransAdjustment;
+    	else if(phase1 && this.isSymptomatic()) transAdjustment = this.symptomaticTransAdjustment;
+    	else if(phase2 && this.isSymptomatic()) transAdjustment = this.symptomaticTransAdjustment;
```
Comment (paulbessell) [inline src/main/java/uk/co/ramp/covid/simulation/covid/Covid.java:211]: phase2 should always be symptomatic, so we could drop the second part there. Although, I probably wrote that bit of code in the first place!

File: src/main/java/uk/co/ramp/covid/simulation/population/Person.java [hunk 2]
```diff
@@ -226,4 +248,27 @@ public Sex getSex() {
     public int getAge() {
         return age;
     }
+
+    public boolean wasTested() {
+        return testOutcome.isPresent();
+    }
+
+    public void getTested() {
+        if (cVirus == null || wasTested() || !cVirus.isSymptomatic()) {
```
Comment (paulbessell) [inline src/main/java/uk/co/ramp/covid/simulation/population/Person.java:257]: What does wasTested() achieve here. Is it sayingt hat if the person has bene tested previously then they can not be tested again. If so thenw e should drop the requirement

File: src/main/java/uk/co/ramp/covid/simulation/population/Population.java [hunk 3]
```diff
@@ -306,6 +306,8 @@ public void timeStep(Time t, DailyStats dStats) {
         // households.parallelStream().forEach(h -> h.doInfect(dStats));
         // places.getAllPlaces().parallelStream().forEach(p -> p.doInfect(dStats));
 
+        households.forEach(h -> h.doTesting(t));
```
Comment (bobtaylor12) [inline src/main/java/uk/co/ramp/covid/simulation/population/Population.java:309]: Are we assuming everyone who is symptomatic is going to be tested?

File: src/main/java/uk/co/ramp/covid/simulation/population/Person.java (hunk 2 above)
Comment (BlairArchibald) [reply src/main/java/uk/co/ramp/covid/simulation/population/Person.java:257]: It's meant to be to only let them get tested twice. If they are allowed to get tested again when should this happen?

Comment (paulbessell) [reply src/main/java/uk/co/ramp/covid/simulation/population/Person.java:257]: My bad. Well, sort of. I was thinking of people being tested when they don't have Covid, but that doesn't apply here. It is possible that people are tested multiple times during a single Covid infection, but it is a reasonable asusmption that they are infected once during infection. 
We may later add a further test for being tested once the patient enters phase 2 but this sin't critical just now

**Pull request merged**